xikolo-ios/VideoViewController.swift

```diff
@@ -15,24 +15,55 @@ class VideoViewController: UIViewController {
     @IBOutlet weak var containerVideoView: UIView!
     @IBOutlet weak var titleLabel: UILabel!
     @IBOutlet weak var descriptionTextView: UITextView!
+    @IBOutlet weak var openSlidesButton: UIButton!
+    @IBOutlet weak var previousItemButton: UIButton!
+    @IBOutlet weak var nextItemButton: UIButton!
+    @IBOutlet weak var containerHeighConstraint: NSLayoutConstraint!
     
-    var courseItem: CourseItem! {
+    var courseItem: CourseItem!
+    var video: Video? = nil {
         didSet {
-            updateUI()
+            videoDidLoad()
         }
     }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        containerHeighConstraint.constant = self.view.frame.size.width * (9/16)
+        let videoIncomplete = courseItem.content as! Video
+        VideoHelper.syncVideo(videoIncomplete).onSuccess { videoComplete in
+            self.video = videoComplete
+            self.performSegueWithIdentifier("EmbedAVPlayer", sender: self.video)
+        }
+    }
+
+    func videoDidLoad() {
+        titleLabel.text = courseItem.title
```

Comment:
Why don't you do this in `viewDidLoad` instead? The `courseItem` is there before the view controller is even loaded, right?